SpatialGDK/Build/Programs/Improbable.Unreal.Scripts/DeploymentLauncher/DeploymentLauncher.cs

```diff
@@ -241,80 +267,111 @@ private static int CreateSimDeployments(string[] args, bool useChinaPlatform)
             var assemblyName = args[2];
             var runtimeVersion = args[3];
             var targetDeploymentName = args[4];
-            var simDeploymentName = args[5];
+            var simDeploymentBaseName = args[5];
             var simDeploymentJson = args[6];
             var simDeploymentRegion = args[7];
             var simDeploymentCluster = args[8];
             var simDeploymentSnapshotPath = args[9];
-
-            var simNumPlayers = 0;
-            if (!Int32.TryParse(args[10], out simNumPlayers))
+            
+            var numSimplayers = 0;
+            if (!Int32.TryParse(args[10], out numSimplayers))
             {
                 Console.WriteLine("Cannot parse the number of simulated players to connect.");
                 return 1;
             }
+            else if (numSimplayers <= 0)
+            {
+                Console.WriteLine("The number of players must be positive.");
+                return 1;
+            }
 
             var autoConnect = false;
-            if (!Boolean.TryParse(args[11], out autoConnect))
+            if (!Boolean.TryParse(args[12], out autoConnect))
             {
                 Console.WriteLine("Cannot parse the auto-connect flag.");
                 return 1;
             }
 
-            try
+            var maxPlayersPerDeployment = numSimplayers;
+            if (args.Length >= 12)
             {
-                var deploymentServiceClient = DeploymentServiceClient.Create(GetApiEndpoint(useChinaPlatform));
-
-                if (DeploymentExists(deploymentServiceClient, projectName, simDeploymentName))
+                if (!Int32.TryParse(args[11], out maxPlayersPerDeployment))
                 {
-                    StopDeploymentByName(deploymentServiceClient, projectName, simDeploymentName);
+                    Console.WriteLine("Cannot parse the target number of simulated players deployments.");
+                    return 1;
                 }
-
-                var createSimDeploymentOp = CreateSimPlayerDeploymentAsync(deploymentServiceClient,
-                    projectName, assemblyName, runtimeVersion, targetDeploymentName, simDeploymentName,
-                    simDeploymentJson, simDeploymentSnapshotPath, simDeploymentRegion, simDeploymentCluster, simNumPlayers, useChinaPlatform);
-
-                // Wait for both deployments to be created.
-                Console.WriteLine("Waiting for the simulated player deployment to be ready...");
-                var simPlayerDeployment = createSimDeploymentOp.PollUntilCompleted().GetResultOrNull();
-                if (simPlayerDeployment == null)
+                else if (maxPlayersPerDeployment <= 0)
                 {
-                    Console.WriteLine("Failed to create the simulated player deployment");
+                    Console.WriteLine("The number of deployments must be positive.");
```

Comment:
isn't this the number of players per deployment?

Reply:
Right - fixed!